portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,75 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private List<String> quotes;
+
+   public void init() {
+    quotes = new ArrayList<>();
+    quotes.add("Here is one message!");
+    quotes.add("Here is a second message!");
+    quotes.add("Here is a third message!");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String json = convertToJson(quotes);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    //response.getWriter().println("<h1>Hello Emily!</h1>");
+  }
+
+    /**
+   * Converts a ServerStats instance into a JSON string using manual String concatentation.
+   */
+   //have to use a for loop, convert ArrayList to json adding keys
+  private String convertToJson(List<String> listt) {
+    String json = "{";
+    for (int i=0; i<listt.size(); i++) {
+        //message 1
+        json +=  "\"" + String.valueOf(i+1) +  "\"";
+        json += ": ";
+        json +=  "\"" + listt.get(i) + "\"";
+        if (i != (listt.size()-1)) {
+        json += ",";
+        }
+    }
+    json += "}" ;
+    return json;
+  }
+
+//Do post for POST 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+
+    // Break the text into individual words.
+    String[] words = text.split("\\s*,\\s*");
```

Comment:
Hey I don't think this step is necessary. Why would you want to break the text into individual words before processing them in doGet()?

All that needs to happen in this doPost() method is for you to get the String text from the form and add it to your `quotes` list of strings! Since you're redirecting the page back to `index.html`, doGet() runs right after doPost() to process the list of `quotes` and displays newly added comment/quote string to the frontend.